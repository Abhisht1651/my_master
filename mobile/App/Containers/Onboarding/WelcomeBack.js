// @flow

import React from 'react'
import { View, Image, Text, Platform } from 'react-native'
import { Images } from '../../Themes'

// Styles
import styles from '../Styles/Onboarding/WelcomeBackStyles'
import { MediumButton } from '../../Components/Buttons/MediumButton'

const WelcomeBack = () => {
  return (
    <View style={styles.section}>
      {Platform.OS === 'ios' && <View style={styles.statusBar} />}
      <Image source={Images.newzeraPurpleLogo} style={styles.logo} />
      <Text style={styles.heading}>Welcome back, Amanda</Text>
      <Image source={Images.bitmap2x} style={styles.profilepic} />
      <Text style={styles.or}>or</Text>
      <View style={styles.button}>
        <MediumButton outline title="Sign up with another account" />
      </View>
    </View>
  )
}
export default WelcomeBack
