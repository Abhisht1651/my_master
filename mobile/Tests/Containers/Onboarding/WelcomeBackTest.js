import { Image } from 'react-native'
import React from 'react'
import { shallow } from 'enzyme'
import WelcomeBack from '../../../App/Containers/Onboarding/WelcomeBack'
import { MediumButton } from '../../../App/Components/Buttons/MediumButton'

const createTestProps = () => ({
  navigation: {
    navigate: jest.fn()
  }
})

describe('WelcomeBack', () => {
  describe('rendering', () => {
    let wrapper
    let props // use type "any" to opt-out of type-checking
    beforeEach(() => {
      props = createTestProps({})
      wrapper = shallow(<WelcomeBack props />) // no compile-time error
      let x = props
      if (x) x = props
    })
    it('should match to snapshot', () => {
      expect(wrapper).toMatchSnapshot()
    })

    it('should render a <RoundedButton />', () => {
      expect(wrapper.find(MediumButton)).toHaveLength(1)
    })
    it('should render 2 <Image />', () => {
      expect(wrapper.find(Image)).toHaveLength(2)
    })
  })
})
