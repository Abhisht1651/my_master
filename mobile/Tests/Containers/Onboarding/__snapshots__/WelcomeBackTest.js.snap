// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomeBack rendering should match to snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "height": 58.188802083333336,
        "width": "100%",
      }
    }
  />
  <Image
    source={
      Object {
        "default": "",
      }
    }
    style={
      Object {
        "height": 217.12239583333334,
        "marginTop": 236.22916666666669,
        "resizeMode": "contain",
        "width": 376.9244791666667,
      }
    }
  />
  <Text
    style={
      Object {
        "color": "#1F1F60",
        "fontFamily": "BananaGrotesk-Medium",
        "fontSize": 39,
        "marginTop": 36.4765625,
      }
    }
  >
    Welcome back, Amanda
  </Text>
  <Image
    source={
      Object {
        "default": "",
      }
    }
    style={
      Object {
        "borderRadius": 83.375,
        "height": 165.01302083333334,
        "marginTop": 67.7421875,
        "resizeMode": "cover",
        "width": 165.01302083333334,
      }
    }
  />
  <Text
    style={
      Object {
        "color": "#1F1F60",
        "fontFamily": "BananaGrotesk-Regular",
        "fontSize": 28,
        "marginTop": 62.53125,
      }
    }
  >
    or
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "marginTop": 90.32291666666667,
      }
    }
  >
    <MediumButton
      disabled={false}
      onPress={[Function]}
      outline={true}
      style={Object {}}
      title="Sign up with another account"
    />
  </View>
</View>
`;
